packages/core/src/CSL/coreToCsl/coreToCsl.ts

```diff
@@ -44,14 +44,29 @@ import { SerializationError } from '../../errors';
 import { SerializationFailure } from '../..';
 import { parseCslAddress } from '../parseCslAddress';
 
-export const tokenMap = (assets: Cardano.TokenMap) => {
+export const tokenMap = (map: Cardano.TokenMap) => {
   const multiasset = MultiAsset.new();
-  for (const assetId of assets.keys()) {
-    const { scriptHash, assetName } = Asset.util.parseAssetId(assetId);
-    const assetsObj = Assets.new();
-    const amount = BigNum.from_str(assets.get(assetId)!.toString());
-    assetsObj.insert(assetName, amount);
-    multiasset.insert(scriptHash, assetsObj);
+  const policyObj: {
+    [policyId: string]: { assetsMap: Map<AssetName, BigNum>; scriptHash: ScriptHash };
+  } = {};
+
+  for (const assetId of map.keys()) {
+    const { assetName, scriptHash } = Asset.util.parseAssetId(assetId);
+    // REVIEW: Could also return the policyId from Asset.util.parseAssetId
```

Comment:
It already returns it as `scriptHash` object. Perhaps it should be exported as `coreToCsl.asset` though :thinking:

Reply:
Captured in ADP-1876